src/server/scripts/Spells/spell_warrior.cpp

```diff
@@ -1078,6 +1080,61 @@ class spell_warr_heroic_leap_jump : public SpellScriptLoader
     }
 };
 
+// Shockwave - 46968
+class spell_warr_shockwave : public SpellScriptLoader
+{
+public:
+    spell_warr_shockwave() : SpellScriptLoader("spell_warr_shockwave") { }
+
+    class spell_warr_shockwave_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_warr_shockwave_SpellScript);
+
+        int32 hitTargets = 0;
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            if (!sSpellMgr->GetSpellInfo(SPELL_WARRIOR_SHOCKWAVE) ||
+                !sSpellMgr->GetSpellInfo(SPELL_WARRIOR_SHOCKWAVE_STUN))
+                return false;
+            return true;
+        }
+
+        void HandleDamage(SpellEffIndex /*effIndex*/)
+        {
+            if (Unit* caster = GetCaster())
```

Comment:
If I recall correctly, caster cannot be null. But always use curly braces if another if block is coming. Some compilers requires it I think.